Set `platform-9` team as the code owner for config

wzieba opened the pull request:
According to [documentation](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners#example-of-a-codeowners-file):

> #Order is important; the last matching pattern takes the most
#precedence. When someone opens a pull request that only
#modifies JS files, only @js-owner and not the global
#owner(s) will be requested for a review.
*.js    @js-owner

and this seems to be our case

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

wzieba committed:
Set `platform-9` team as the code owner for config

wzieba commented on .github/CODEOWNERS:
```diff
@@ -2,17 +2,17 @@
-.ruby-version       @woocommerce/android-developers @woocommerce/platform-9
-Gemfile             @woocommerce/android-developers @woocommerce/platform-9
-Gemfile.lock        @woocommerce/android-developers @woocommerce/platform-9
-*.gradle            @woocommerce/android-developers @woocommerce/platform-9
```
I've also made this rule more strict to exclude `build.gradle` files. They do contain build-related config but the most often case (and quite often) is to just update dependency version and I don't want us to flood `platform-9` with this kind of PRs.

AliSoftware commented:
Conceptually OK but we'll change the team to `@owl-team` soon, so we should wait for this change of team to be done in GH before merging this PR.

AliSoftware commented:
@wzieba The GitHub team change has finally been made! – paaHJt-2jX-p2

I'll let you update the PR accordingly to make `CODEOWNERS` ping @woocommerce/owl-team instead, and ask me/us for review again once your changes are done 🙇

wzieba committed:
Set Owl Team as build tools code owners

wzieba commented:
@AliSoftware thanks! I've updated the file and also updated Owl Team permissions to the repository as it's required:

> The people you choose as code owners must have write permissions for the repository. When the code owner is a team, that team must have write permissions, even if all the individual members of the team already have write permissions directly, through organization membership, or through another team membership.

[Source](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners)

AliSoftware commented on .github/CODEOWNERS:
```diff
@@ -2,17 +2,17 @@
-gradlew             @woocommerce/android-developers @woocommerce/platform-9
-gradlew.bat         @woocommerce/android-developers @woocommerce/platform-9
+/.bundle/           @woocommerce/owl-team
+/.circleci/         @woocommerce/owl-team
```
Even if it does not exist _yet_, I'm wondering if we shouldn't add `/.buildkite/` here too in anticipation, given we're about to migrate from CircleCI to BuildKite (example of [how this folder looks like in WPAndroid for comparison](https://github.com/wordpress-mobile/WordPress-Android/tree/develop/.buildkite))?

wzieba replied:
Sure, let's try it - [I've tested it on a test fork](https://github.com/wzieba/woocommerce-android/blob/develop/.github/CODEOWNERS) and I'm still recognized there as a codeowner. I've checked it because I was thinking that maybe "invalid syntax" refers also to pointing to not existing directories but it seems it doesn't:

> If any line in your CODEOWNERS file contains invalid syntax, the file will not be detected and will not be used to request reviews.

[Source](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners#codeowners-syntax)

AliSoftware replied:
I love your attention to details and thinking about edge cases + your way of quoting the doc about it 👌 👏

wzieba committed:
Add buildkite directory to codeowners
